### PR TITLE
Add PermanentRestriction tests

### PR DESCRIPTION
**Pull request by Yberjon**
# What ❔

This PR adds tests for PermanentRestriction contract.

### CODE REVIEW

**Review by saxenism — APPROVED** [review]
It would be great if we could test the (relevant and then random) values of the public mappings that are affected by calling certain functions.

Rest looks fine.

File: l1-contracts/test/foundry/unit/concrete/governance/Governance/PermanentRestriction.t.sol [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+        });
+
+        StateTransitionManagerInitializeData memory stmInitializeDataNoGovernor = StateTransitionManagerInitializeData({
+            owner: address(0),
+            validatorTimelock: validator,
+            chainCreationParams: chainCreationParams,
+            protocolVersion: 0
+        });
+
+        vm.expectRevert(bytes.concat("STM: owner zero"));
```
Comment (StanislavBreadless) [inline l1-contracts/test/foundry/unit/concrete/governance/Governance/PermanentRestriction.t.sol:129]: why do we have this test in the setUp?

File: l1-contracts/test/foundry/unit/concrete/governance/Governance/PermanentRestriction.t.sol [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+        chainAdmin = new ChainAdmin(restrictions);
+    }
+
+    function test_ownerAsAddressZero() public {
+        vm.expectRevert(ZeroAddress.selector);
+        permRestriction = new PermanentRestriction(address(0), iBridgehub);
+    }
+
+    function test_allowAdminImplementation(bytes32 implementationHash) public {
+        vm.expectEmit();
```
Comment (StanislavBreadless) [inline l1-contracts/test/foundry/unit/concrete/governance/Governance/PermanentRestriction.t.sol:178]: topics could be compared

File: l1-contracts/test/foundry/unit/concrete/governance/Governance/PermanentRestriction.t.sol [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+            _chainId: chainId,
+            _stateTransitionManager: address(chainContractAddress),
+            _baseToken: baseToken,
+            _salt: 0,
+            _admin: newChainAdmin,
+            _initData: abi.encode(_diamondCut)
+        });
+    }
+
+    function gettersSelectors() public pure returns (bytes4[] memory) {
```
Comment (StanislavBreadless) [inline l1-contracts/test/foundry/unit/concrete/governance/Governance/PermanentRestriction.t.sol:346]: can't these be reused from other files? creating a chain seems like a common task

Comment (Yberjon) [reply l1-contracts/test/foundry/unit/concrete/governance/Governance/PermanentRestriction.t.sol:346]: I got rid off getGettersSelectors and added missing selector to utils function.
Regarding createNewChain i have not found a function that would use bridgehubs create new chain.
I can reuse getDiamondCutData function from StateTransitionManagerTest but would need to rename its setup function and make adjustments in tests inheriting from it.

Comment (StanislavBreadless) [reply l1-contracts/test/foundry/unit/concrete/governance/Governance/PermanentRestriction.t.sol:346]: okay

**Review by StanislavBreadless — APPROVED** [review]

**Comment by StanislavBreadless** [comment]
I am not sure why coverage fails, I'll assume it is some CI bug and will merge

**Pull request merged**